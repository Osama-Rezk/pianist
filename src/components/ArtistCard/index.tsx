import React from "react";
import styled from "@emotion/styled";
import { Artist } from "../../types";

const Container = styled.figure({
  display: "flex",
  flexDirection: "column",
  alignItems: "center",
});

const Description = styled.h1({
  color: "#191919",
  fontSize: 20,
});

const Image = styled.img({
  height: 264,
  borderRadius: "50%",
  width: 264,
  boxShadow: "0 1px 6px rgba(25,25,34,.16)",
  marginBottom: 5,
});

export const ArtistCard: React.FC<Artist> = (props) => {
  const { name, picture } = props;
  return (
    <Container>
      <Image alt={`a picture of artist ${name}`} id="artistpic" src={picture} />
      <Description id="artistname">{name}</Description>
    </Container>
  );
};
