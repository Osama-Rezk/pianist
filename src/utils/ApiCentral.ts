import axios, { AxiosRequestConfig } from "axios";
import { baseURL } from "../constants/config";

export async function request(options: AxiosRequestConfig) {
  const client = axios.create({
    baseURL,
  });

  const onSuccess = function (response: any) {
    return response.data;
  };

  const onError = function (error: any) {
    return Promise.reject(error.response || error.message);
  };

  return client(options).then(onSuccess).catch(onError);
}
