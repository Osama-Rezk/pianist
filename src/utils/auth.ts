export const getJwt = () => localStorage.getItem("access_token");
