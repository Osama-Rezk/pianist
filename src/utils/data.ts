import { Genre, Artist, UserState } from "../types";

export const genreList: Genre[] = [
  {
    id: 1,
    name: "POP",
    picture: "",
    picture_big: "",
    picture_medium: "",
    picture_small: "",
    picture_xl: "",
    type: "",
  },
  {
    id: 1,
    name: "R & R",
    picture: "",
    picture_big: "",
    picture_medium: "",
    picture_small: "",
    picture_xl: "",
    type: "",
  },
];

export const artistsList: Artist[] = [
  {
    id: 1,
    name: "Artist 1",
    picture: "",
    picture_big: "",
    picture_medium: "",
    picture_small: "",
    picture_xl: "",
    type: "",
    radio: false,
    tracklist: "",
  },
  {
    id: 2,
    name: "Artist 2",
    picture: "",
    picture_big: "",
    picture_medium: "",
    picture_small: "",
    picture_xl: "",
    type: "",
    radio: false,
    tracklist: "",
  },
];

export const user: UserState = {
  name: "user",
  id: 1,
  lastname: "",
  firstname: "",
  email: "",
  status: 1,
  birthday: "",
  inscription_date: "",
  gender: "",
  picture: "",
  type: "",
  picture_big: "",
  picture_medium: "",
  picture_small: "",
  picture_xl: "",
  country: "",
  is_kid: false,
};
