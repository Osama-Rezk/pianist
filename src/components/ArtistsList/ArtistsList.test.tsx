import React from "react";
import { ArtistsList } from "./";
import { mount } from "enzyme";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import { ArtistCard } from "../ArtistCard";
import { artistsList } from "../../utils/data";
const mockStore = configureStore([thunk]);

describe("ArtistsList Component", () => {
  let store = mockStore({
    genres: {
      data: [],
      artists: artistsList,
    },
  });

  const getWrapper = () =>
    mount(
      <Provider store={store}>
        <ArtistsList />
      </Provider>
    );

  it("should render my component", () => {
    const wrapper = getWrapper();

    expect(wrapper.exists()).toBe(true);
  });

  it("should render correctly", () => {
    const wrapper = getWrapper();

    expect(wrapper).toMatchSnapshot();
  });

  it("it should not return null", () => {
    const wrapper = getWrapper();

    expect(wrapper.isEmptyRender()).toBe(false);
  });

  it("should render 2 elemnts in store  ", () => {
    const wrapper = getWrapper();

    expect(wrapper.find(ArtistCard).length).toEqual(2);
  });
});
