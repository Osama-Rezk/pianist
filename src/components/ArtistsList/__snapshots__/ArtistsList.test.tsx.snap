// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtistsList Component should render correctly 1`] = `ReactWrapper {}`;

exports[`Button Component should render correctly 1`] = `ReactWrapper {}`;
